### PR TITLE
[Table sanity check] sanity check for the table interface_addresses

### PR DESCRIPTION
**Pull request by akindyakov**
Issue: #5055

### CODE REVIEW

File: osquery/tables/networking/posix/interfaces.cpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ void genAddressesFromAddr(const struct ifaddrs* addr, QueryData& results) {
       r["point_to_point"] = dest_address;
     }
   }
-
+  r["type"] = "unknown";
```
Comment (fmanco) [inline osquery/tables/networking/posix/interfaces.cpp]: Need to update spec file which doesn't have unknown.

File: osquery/tests/integration/tables/interface_addresses.cpp [hunk 2]
```diff
@@ -12,33 +12,37 @@
...
+  auto const verifyEmptyStringOrIpAddress = [](auto const& value) {
+    return value.empty() ? true : verifyIpAddress(value);
+  };
+  auto const row_map = ValidatatioMap{
+      {"interface", NonEmptyString},
+      {"address", verifyIpAddress},
+      {"mask", verifyIpAddress},
+      {"broadcast", verifyEmptyStringOrIpAddress},
+      {"point_to_point", verifyEmptyStringOrIpAddress},
+      {"type", SpecificValuesCheck{"manual", "dhcp", "auto", "unknown"}},
```
Comment (fmanco) [inline osquery/tests/integration/tables/interface_addresses.cpp]: spec files says `other` is also possible.

File: osquery/tables/networking/posix/interfaces.cpp (hunk 1 above)
Comment (guliashvili) [reply osquery/tables/networking/posix/interfaces.cpp]: maybe we could use other instead of unknown

Comment (akindyakov) [reply osquery/tables/networking/posix/interfaces.cpp]: Agree with updating spec file, thanks.

Do not agree with using `other` for `unknown` for two reasons:
1. By meaning `other` means we know the type of interface, but the one is differ from `dhcp`, `auto`, `manual`.
2. Windows is already uses the word `unknown` see implementation for the windows: [ref](https://github.com/facebook/osquery/blob/master/osquery/tables/networking/windows/interfaces.cpp#L202).

**Pull request merged**